In [4]:
import pandas as pd
from IPython.display import display

In [5]:
# load file csv-nya
df = pd.read_csv('../datasets/data.csv', index_col=0)

In [13]:
# cek apakah data berupa numerik
numeric_columns = df.select_dtypes(include=['number']).shape[1]
total_columns = df.shape[1]

if numeric_columns == total_columns:
    data_type = "Numeric"
else:
    error_columns = total_columns - numeric_columns
    data_type = f"Ada Error! ({numeric_columns} Numeric, {error_columns} Non-Numeric)"

In [14]:
# validasi jumlah kolom dan sampel
summary = pd.DataFrame({
    "Dimensi": ["Rows (Samples)", "Columns (Features)", "Data Type"],
    "Data": [df.shape[0], df.shape[1], data_type]
})

In [15]:
# hasil harus berupa numerik
display(summary)

,Dimensi,Data
0,Rows (Samples),801
1,Columns (Features),20531
2,Data Type,Numeric


Hasil validasi menunjukkan terdapat 801 sampel dan 20531 fitur serta data yang dimiliki semuanya berupa numerik. Ini menandakan dataset ini bisa digunakan untuk kebutuhan topik skripsi yang saya angkat saat ini mengingat algoritma K-means bekerja dengan sangat baik dengan data numerik karena algortima K-means menggunakan Euclidean Distance untuk mengukur jarak antar data.